common/src/test/java/com/redhat/cloud/notifications/models/event/TestEventHelperTest.java

```diff
@@ -0,0 +1,197 @@
+        final var expectedRecipientsCount = 1;
+        final var recipients = testAction.getRecipients();
+
+        Assertions.assertEquals(expectedRecipientsCount, recipients.size(), "unexpected number of recipients in the test action");
+
+        final var recipient = recipients.get(0);
+        final var users = recipient.getUsers();
+
+        Assertions.assertEquals(expectedRecipientsCount, users.size(), "unexpected number of test action users");
+
+        final var user = users.get(0);
+
+        Assertions.assertEquals(TestEventHelper.TEST_ACTION_RECIPIENT, user, "unexpected user in the test action");
+
+        // Check the events, their metadata and their payload.
+        final var events = testAction.getEvents();
+
+        final var expectedEventsCount = 1;
+        Assertions.assertEquals(expectedEventsCount, events.size(), "unexpected number of test action events");
+
+        final Event event = events.get(0);
+        final Metadata metadata = event.getMetadata();
+        final Map<String, Object> metaAdditionalProperties = metadata.getAdditionalProperties();
+
+        final var expectedMetadataAdditionalPropertiesCount = 1;
+        Assertions.assertEquals(expectedMetadataAdditionalPropertiesCount, metaAdditionalProperties.size(), "unexpected number of metadata additional properties");
+
+        final String metadataValue = (String) metaAdditionalProperties.get(TestEventHelper.TEST_ACTION_METADATA_KEY);
+        Assertions.assertEquals(TestEventHelper.TEST_ACTION_METADATA_VALUE, metadataValue, "unexpected event metadata value");
+
+        final Payload payload = event.getPayload();
+        final Map<String, Object> payloadAdditionalProperties = payload.getAdditionalProperties();
+
+        final var expectedPayloadAdditionalPropertiesCount = 1;
+        Assertions.assertEquals(expectedPayloadAdditionalPropertiesCount, payloadAdditionalProperties.size(), "unexpected number of payload additional properties");
+
+        final String payloadValue = (String) payload.getAdditionalProperties().get(TestEventHelper.TEST_ACTION_PAYLOAD_KEY);
+
+        Assertions.assertEquals(TestEventHelper.TEST_ACTION_PAYLOAD_VALUE, payloadValue, "unexpected event payload value");
+    }
+
+    /**
+     * Tests that a test integration event is properly identified as such by
+     * the function under test.
+     */
+    @Test
+    public void testIsIntegrationEvent() {
+        final Action testAction = TestEventHelper.createTestAction(UUID.randomUUID(), "random-org-id");
+
+        final var testEvent = new com.redhat.cloud.notifications.models.Event();
+        testEvent.setAction(testAction);
+
+        Assertions.assertTrue(TestEventHelper.isIntegrationTestEvent(testEvent), "the test event was not identified as such");
+    }
+
+    /**
+     * Tests that when an event has no context, the function under test
+     * correctly identifies it as a normal event.
+     */
+    @Test
+    public void testIsIntegrationEventNoContext() {
```

Comment:
The test will fail because the B/A/ET do not match the expectations from the tested method, not because the event has no context.

Reply:
You are right. I have removed the test, since it doesn't make any sense to have it around...